pkgs/by-name/me/melos/package.nix

```diff
@@ -0,0 +1,40 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildDartApplication,
+}:
+let
+  pname = "melos";
+  version = "6.1.0";
+  src = fetchFromGitHub {
+    owner = "invertase";
+    repo = pname;
+    rev = "melos-v${version}";
+    hash = "sha256-QebhZpPK2OBRUH/7cz8g9WGgKiYuFznvBp4xrRjSpVA=";
+  };
+in
+buildDartApplication {
+  inherit pname version src;
+
+  sourceRoot = "${src.name}/packages/melos";
+
+  patches = [ ./add-generic-main.patch ];
+
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  # hard code the path to the melos templates
+  preBuild = ''
+    substituteInPlace lib/src/common/utils.dart --replace-fail final\ melosPackageFileUri\ =\ await\ Isolate.resolvePackageUri\(melosPackageUri\)\; return\ \"$out\"\;
+    substituteInPlace lib/src/common/utils.dart --replace-fail return\ p.normalize\(\'\$\{melosPackageFileUri!.toFilePath\(\)}/../..\'\)\; " "
```

Comment:
Can you please wrap with double quotes the string and separate them correctly. Reading this is very complicated, it's hard to distinguish what is going to be replaced.